### ISSUE TITLE
Gradient of self.mnemonics is None after backward in mnemonics_train.py

### ISSUE THREAD

**Issue by cyberknight42** [yaoyao-liu/class-incremental-learning#5]
Hi,

I run the experiment with the following command `"python main.py --nb_cl_fg=50 --nb_cl=2 --nb_protos 20 --resume --imprint_weights"`. I find that the gradient of `self.mnemonics` in mnemonics_train.py is _None_ after `q_loss.backward()`, which means it doesn't update.
I am wondering whether this is correct?

Thanks.

**Comment by yaoyao-liu** [comment]
Hi,

The gradient for `self. mnemonics` should not be `None`. Thanks for reporting this issue. I'll check if there is any bug.

Best,
Yaoyao

**Comment by yaoyao-liu** [comment]
Hi,

Thanks for reporting this issue.
The gradients for `self.mnemonics` should not be `None`. I'll check if there is any bug. 

Best,
Yaoyao
